platform/machine/aws/flight.go

```diff
@@ -68,6 +69,8 @@ func NewFlight(opts *aws.Options) (platform.Flight, error) {
 		return nil, err
 	}
 
+	af.keyAdded = true
```

Comment:
Please snug this up in the same block as the `AddKey` call.  As is, it's too easy for another code block to be inserted between them.